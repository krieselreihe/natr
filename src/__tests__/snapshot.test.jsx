const React = require("react");
const renderer = require("react-test-renderer");

const { describe, toMatchSnapshot } = require("../runner");

describe("toMatchSnapshot", (assert) => {
  assert({
    given: "an object",
    should: "match a specific snapshot",
    actual: { id: 1, name: "Mary" },
    expected: toMatchSnapshot(),
  });

  assert({
    given: "a different object",
    should: "match a specific snapshot",
    actual: { id: 2, name: "Martin", active: true },
    expected: toMatchSnapshot(),
  });

  assert({
    given: "a React component",
    should: "match a specific snapshot",
    actual: renderer.create(<div>Hello World</div>).toJSON(),
    expected: toMatchSnapshot(),
  });
});
