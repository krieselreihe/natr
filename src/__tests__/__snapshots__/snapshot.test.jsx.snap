{
  "toMatchSnapshot, Given an object: should match a specific snapshot": {
    "id": 1,
    "name": "Mary"
  },
  "toMatchSnapshot, Given a different object: should match a specific snapshot": {
    "id": 2,
    "name": "Martin",
    "active": true
  },
  "toMatchSnapshot, Given a React component: should match a specific snapshot": {
    "type": "div",
    "props": {},
    "children": [
      "Hello World"
    ]
  }
}